cg/services/orders/order_status_service/utils.py

```diff
@@ -0,0 +1,86 @@
+from enum import Enum
+from cg.apps.tb.dto.summary_response import AnalysisSummary
+from cg.services.orders.order_status_service.dto.order_status_summary import OrderSummary
+from cg.services.orders.order_status_service.dto.case_status_summary import CaseSummary
+from cg.store.models import Case, Order
+
+
+def create_summaries(
+    case_summaries: list[CaseSummary], analysis_summaries: list[AnalysisSummary]
+) -> list[OrderSummary]:
+    summaries: list[OrderSummary] = initialise_order_summaries(case_summaries)
+    add_analysis_summaries(summaries, analysis_summaries)
+    return summaries
+
+
+def initialise_order_summaries(case_summaries: list[CaseSummary]) -> list[OrderSummary]:
+    summaries: list[OrderSummary] = []
+    for case_summary in case_summaries:
+        summary: OrderSummary = OrderSummary(
+            order_id=case_summary.order_id,
+            total_cases=case_summary.total,
+            in_sequencing=case_summary.in_sequencing,
+            in_preparation=case_summary.in_lab_preparation,
+        )
+        summaries.append(summary)
+    return summaries
+
+
+def add_analysis_summaries(
+    order_summaries: list[OrderSummary],
+    analysis_summaries: list[AnalysisSummary],
+) -> None:
+    order_summary_map = {summary.order_id: summary for summary in order_summaries}
+    for analysis_summary in analysis_summaries:
+        order_summary = order_summary_map[analysis_summary.order_id]
+        order_summary.delivered = analysis_summary.delivered
+        order_summary.running = analysis_summary.running
+        order_summary.cancelled = analysis_summary.cancelled
+        order_summary.failed = analysis_summary.failed
```

Comment:
Just so that I understand our assumptions. If a case would have an analysis i trailblazer but one of its samples would not have a last_sequenced at, it would show up twice in our count. I think that assumption is safe but it's good to be aware of it

Reply:
Good observation!
Is that a possible state? That a sample has not been sequenced but an analysis is running for it already?
Sounds like it should not be allowed

Reply:
Another issue is re-sequencing. Even if a sample really is in sequencing, it will not show up as such

Reply:
We will probably revisit this - closing for now.